executor/insert_common.go

```diff
@@ -198,11 +201,12 @@ func (e *InsertValues) getRow(cols []*table.Column, list []expression.Expression
 
 	for i, expr := range list {
 		val, err := expr.Eval(chunk.MutRowFromDatums(row).ToRow())
-		if err = e.handleErr(cols[i], rowIdx, err); err != nil {
+		valStr, _ := val.ToString()
```

Comment:
Could you move the ToString() operation into the handErr() function? We could use a lazy way to call the ToString function. Or you will call ToString for all the data.

Reply:
done.